firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -351,6 +397,14 @@ public synchronized void beginRealtimeHttpStream() {
       return;
     }
 
+    ConfigMetadataClient.RealtimeBackoffMetadata backoffMetadata =
+        metadataClient.getRealtimeBackoffMetadata();
+    Date currentTime = new Date(clock.currentTimeMillis());
+    if (currentTime.before(backoffMetadata.getBackoffEndTime())) {
```

Comment:
Wouldn't we want to wait to retry if the backoff end time is in the future? Maybe I'm misreading this?

Reply:
backoffMetadata,getBackoffEndTime() is the backoff time for the Http stream so if the currentTime is before that then we should retry the connection later in the future